File: LeetCode/java/2442_Count_Number_of_Distinct_Integers_After_Reverse_Operations.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int countDistinctIntegers(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for (int n : nums) {
            set.add(n);

            String s = String.valueOf(n);

            StringBuilder sb = new StringBuilder(s);

            sb.reverse();

            set.add(Integer.parseInt(sb.toString()));
        }

        return set.size();
    }
}

File: LeetCode/cpp/2442_Count_Number_of_Distinct_Integers_After_Reverse_Operations.cpp
class Solution {
public:
	int countDistinctIntegers(vector<int>& nums) {
		set<int> st;

		for (int n : nums) {
			st.insert(n);

			string s = to_string(n);

			reverse(s.begin(), s.end());

			st.insert(stoi(s));
		}

		return st.size();
	}
};